content/spin/plugin-authoring.md

```diff
@@ -0,0 +1,105 @@
+---
+
+Spin plugins add new functionality or subcommands to Spin without modifying the
+Spin codebase. They make Spin easily extensible while keeping it lightweight.
+Spin plugins can add new triggers to Spin (such as the [example timer
+trigger](https://github.com/fermyon/spin/blob/main/examples/spin-timer/trigger-timer.json)),
+enable new language SDKs (such as
+[`js2wasm`](https://github.com/fermyon/spin-plugins/blob/main/manifests/js2wasm/js2wasm.json)),
+and more.
+
+This document will cover what Spin plugins are, how to use a plugin, and how to
+create a plugin.
+
+## What are Spin Plugins?
+
+A Spin plugin is an executable that is added to Spin's plugins directory
+(`$XDG_DATA_HOME/spin/plugins`) upon a `spin plugins install <plugin-name>`. The
+plugin is then ready to be used. If the plugin is an extension to the Spin CLI,
+it can now be executed directly as a subcommand: `spin <plugin-name>`. If the
+plugin is a trigger plugin, it will be executed during `spin up` when an app
+using that trigger is run. 
+
+While for now plugins are assumed to be executables, in the future, support for
+plugging in WebAssembly modules may be desirable.
+
+## How to Find and Use a Spin Plugin
+
+Spin maintains a centralized catalogue of available Spin plugins in the [Spin
+plugins repository](https://github.com/fermyon/spin-plugins). During plugin
+installation, if it does not already exist, Spin fetches the remote catalogue
+and creates a local snapshot. To ensure that the local snapshot is up to date,
+it is best to run `spin plugins update` before installing any plugins. 
+
+To list available plugins, run `spin plugins list`. Now, decide which plugin to
+install. For example, the `js2wasm` plugin, which is needed in order to build
+JavaScript Spin applications, can be installed by running:
+
+```sh
+spin plugins install js2wasm
+```
+
+With the plugin installed, you can now call `spin js2wasm` to run it. In this
+case, for example, you might call it from your JavaScript application's npm
+build script. Learn more about building Spin components in JavaScript
+[here](https://developer.fermyon.com/spin/javascript-components.md).
+
+To upgrade installed plugins to newer versions, run `spin plugin update` to
+fetch the latest plugins to the local catalogue and `spin plugin upgrade` to perform the
+upgrade on the installed plugins.
+
+## Authoring a Spin Plugin
+
+Spin plugins are implemented as a manifest that points to one or more `.tar.gz` archives which contain the plugin executables. So, to create a plugin you must:
+
+1. Create tar archives of the executables for the platforms you want to support
+2. Compose a manifest that describes the plugin and lists the URLs for those tar archives
+
+### Packaging a Plugin
+
+After creating your plugin executable, package it along with its license as a
+`tar.gz` archive. Note that the `name` field in the plugin manifest must match
+both the binary and license name (i.e. `$name.license`). See the [`spin-plugins`
```

Comment:
The name must match the license name?  There's a `name.license` field?  Maybe an editing collision here?

Reply:
This is confusing. I was giving an example that the license file would be called `plugin-name.license` but removed the example.

Reply:
Whoa, I didn't know the license file had to have a special name.  I agree to skip it here though.